src/libbson/fuzz/make-dicts.py

```diff
@@ -0,0 +1,313 @@
+
+
+Undefined: Final = UndefinedType()
+ValueType = (
+    int
+    | str
+    | float
+    | None
+    | UndefinedType
+    | Binary
+    | Regex
+    | Doc
+    | CodeWithScope
+    | String
+    | Symbol
+    | Code
+    | OID
+    | DBPointer
+)
+
+
+def encode_value(val: ValueType) -> bytes:
+    match val:
+        case int(n):
+            return struct.pack("<i", n)
+        case str(s):
+            return make_string(s)
+        case float(f):
+            return struct.pack("<d", f)
+        case (
+            Doc()
+            | Binary()
+            | Regex()
+            | UndefinedType()
+            | CodeWithScope()
+            | Code()
+            | String()
+            | Symbol()
+            | DBPointer()
+            | OID()
+        ) as d:
+            return d.__bytes__()
+        case None:
+            return b""
+
+
+class Tag(enum.Enum):
+    EOD = 0
+    Double = 1
+    UTF8 = 2
+    Doc = 3
+    Binary = 5
+    Undefined = 6
+    OID = 7
+    Null = 10
+    Regex = 11
+    DBPointer = 12
+    Code = 13
+    Symbol = 14
+    CodeWithScope = 15
+    Int32 = 16
+    Int64 = 18
+
+
+def element(key: str, value: ValueType, *, type: None | Tag = None) -> bytes:
+    if type is not None:
+        return flatten([bytes([type.value]), make_cstring(key), encode_value(value)])
+
+    match value:
+        case int():
+            type = Tag.Int32
+
+        case float():
+            type = Tag.Double
+        case None:
+            type = Tag.Null
+        case UndefinedType():
+            type = Tag.Undefined
+        case Binary():
+            type = Tag.Binary
+        case Regex():
+            type = Tag.Regex
+        case Doc():
+            type = Tag.Doc
+        case CodeWithScope():
+            type = Tag.CodeWithScope
+        case Code():
+            type = Tag.Code
+        case Symbol():
+            type = Tag.Symbol
+        case str() | String():  # Must appear after Code()/Symbol()
+            type = Tag.UTF8
+        case OID():
+            type = Tag.OID
+        case DBPointer():
+            type = Tag.DBPointer
+    return element(key, value, type=type)
+
+
+class Entry(NamedTuple):
+    key: str
+    "The key for the entry. Only for human readability"
+    value: bytes
+    "The arbitrary bytes that make up the entry"
+
+
+class Comment(NamedTuple):
+    txt: str
+
+
+class Line(NamedTuple):
+    txt: str = ""
+
+
+LineItem = Entry | Comment | Line
+
+
+def escape(b: bytes) -> Iterable[str]:
+    s = b.decode("ascii", "backslashreplace")
+    for u8 in b:
+        try:
+            s = chr(u8)
+            s.encode("ascii")  # Try to encode as ASCII
+        except ValueError:
+            # Byte is not valid UTF-8, or not valid ASCII
+            pass
+        else:
+            if s.isprintable():
+                yield s
+                continue
+        # Byte is not valid ASCII, or is not a printable char
+        yield f"\\x{u8:0>2x}"
```

Comment:
Suggested simplification:

```suggestion
def escape(b: bytes) -> Iterable[str]:
    for u8 in b:
        s = chr(u8)
        if s.isascii() and s.isprintable():
            yield s
            continue
        # Byte is not valid ASCII, or is not a printable char
        yield f"\\x{u8:0>2x}"
```